symcrypt/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+use symcrypt_sys;
+use core::ffi::c_void;
+use std::{ptr};
+use std::mem;
+
+pub const SHA256_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+
+pub struct SymCryptInit;
+impl SymCryptInit {
+    pub fn new () {
+        unsafe { 
+            //symcrypt_sys::SymCryptInit(); 
+            // TODO: Find out why SymCryptInit() breaks on linux / windows (BREAKING)
+        }
+    }
+}
+
+pub struct Sha256State {
+    state: symcrypt_sys::_SYMCRYPT_SHA256_STATE,
+    is_dirty: bool
+} 
+
+impl Sha256State { // Sha256State 
+    pub fn new() -> Self {
+            let mut instance = Sha256State {
+                state: symcrypt_sys::_SYMCRYPT_SHA256_STATE::default(),
+                is_dirty: false
+            };
+            unsafe {
+                symcrypt_sys::SymCryptSha256Init(&mut instance.state);
+            }
+        instance
+    }
+
+    pub fn append(&mut self, data: &[u8] ) {
+        unsafe {
+            symcrypt_sys::SymCryptSha256Append(
+                &mut self.state, // pState
+                data.as_ptr(), // pbData
+                data.len() as symcrypt_sys::SIZE_T, //cbData
+            );
+        }
+        self.is_dirty = true;

```

Comment:
>self.is_dirty = true; [](http://example.com/codeflow?start=8&length=21)

Did you run through any rust pretty formatter?

For instance, do you need the trailing ";" for the last statement?

Reply:
Did not run through a prettier formatter, I will see if I can include one in the pipeline for commits on github